auth/user_mgt.go

```
@@ -0,0 +1,711 @@
+	preparePayload()
+}
+type commonParams struct {
+	payload map[string]interface{}
+	errors  []string
+}
+
+// UserToCreate is the parameter struct for the CreateUser function.
+type UserToCreate struct {
+	commonParams
+}
+
+// UserToUpdate is the parameter struct for the UpdateUser function.
+type UserToUpdate struct {
+	commonParams
+}
+
+type getUserResponse struct {
+	RequestType string               `json:"kind,omitempty"`
+	Users       []responseUserRecord `json:"users,omitempty"`
+}
+
+type responseUserRecord struct {
+	UID                string      `json:"localId,omitempty"`
+	DisplayName        string      `json:"displayName,omitempty"`
+	Email              string      `json:"email,omitempty"`
+	PhoneNumber        string      `json:"phoneNumber,omitempty"`
+	PhotoURL           string      `json:"photoUrl,omitempty"`
+	CreationTimestamp  int64       `json:"createdAt,string,omitempty"`
+	LastLogInTimestamp int64       `json:"lastLoginAt,string,omitempty"`
+	ProviderID         string      `json:"providerId,omitempty"`
+	CustomClaims       string      `json:"customAttributes,omitempty"`
+	Disabled           bool        `json:"disabled,omitempty"`
+	EmailVerified      bool        `json:"emailVerified,omitempty"`
+	ProviderUserInfo   []*UserInfo `json:"providerUserInfo,omitempty"`
+	PasswordHash       string      `json:"passwordHash,omitempty"`
+	PasswordSalt       string      `json:"salt,omitempty"`
+	ValidSince         int64       `json:"validSince,string,omitempty"`
+}
+
+type listUsersResponse struct {
+	RequestType string               `json:"kind,omitempty"`
+	Users       []responseUserRecord `json:"users,omitempty"`
+	NextPage    string               `json:"nextPageToken,omitempty"`
+}
+
+// CreateUser creates a new user with the specified properties.
+func (c *Client) CreateUser(ctx context.Context, params *UserToCreate) (*UserRecord, error) {
+	if params == nil {
+		params = &UserToCreate{}
+	}
+	params.preparePayload()
+	if len(params.errors) > 0 {
```

Please return an error from preparePayload(). Saving errors in the state of struct is not a very good pattern.